crates/router/src/core/payments/helpers.rs

```diff
@@ -5406,6 +5413,539 @@ impl ApplePayData {
     }
 }
 
+pub(crate) const SENDER_ID: &[u8] = b"Google";
+pub(crate) const PROTOCOL: &str = "ECv2";
+
+// Structs for keys and the main decryptor
+pub struct GooglePayTokenDecryptor {
+    root_signing_keys: Vec<GooglePayRootSigningKey>,
+    recipient_id: Option<masking::Secret<String>>,
+    private_key: PKey<openssl::pkey::Private>,
+}
+
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EncryptedData {
+    signature: String,
+    intermediate_signing_key: IntermediateSigningKey,
+    protocol_version: GooglePayProtocolVersion,
+    #[serde(with = "common_utils::custom_serde::json_string")]
+    signed_message: GooglePaySignedMessage,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GooglePaySignedMessage {
+    #[serde(with = "common_utils::Base64Serializer")]
+    encrypted_message: masking::Secret<Vec<u8>>,
+    #[serde(with = "common_utils::Base64Serializer")]
+    ephemeral_public_key: masking::Secret<Vec<u8>>,
+    #[serde(with = "common_utils::Base64Serializer")]
+    tag: masking::Secret<Vec<u8>>,
+}
+
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IntermediateSigningKey {
+    signed_key: masking::Secret<String>,
+    signatures: Vec<masking::Secret<String>>,
+}
+
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GooglePaySignedKey {
+    key_value: masking::Secret<String>,
+    key_expiration: String,
+}
+
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GooglePayRootSigningKey {
+    key_value: masking::Secret<String>,
+    key_expiration: String,
+    protocol_version: GooglePayProtocolVersion,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq)]
+pub enum GooglePayProtocolVersion {
+    #[serde(rename = "ECv2")]
+    EcProtocolVersion2,
+}
+
+// Check expiration date validity
+fn check_expiration_date_is_valid(
+    expiration: &str,
+) -> CustomResult<bool, errors::GooglePayDecryptionError> {
+    let expiration_ms = expiration
+        .parse::<i128>()
+        .change_context(errors::GooglePayDecryptionError::InvalidExpirationTime)?;
+    let expiration_time =
+        time::OffsetDateTime::from_unix_timestamp_nanos(expiration_ms * 1_000_000)
+            .change_context(errors::GooglePayDecryptionError::InvalidExpirationTime)?;
```

Comment:
Can add a comment here as to why the multiplication by `1_000_000`: to obtain time in terms of nanoseconds from a value in milliseconds.